R/bfastlite.R

```diff
@@ -24,18 +24,26 @@
 #' @param ... Additional arguments to \code{\link[strucchange]{breakpoints}}.
 #' @inheritParams bfastpp
 #' @inheritParams strucchange::breakpoints
-#' @example examples/bfast0n.r
+#' @return An object of class \code{bfastlite}, with two elements:
+#' \item{breakpoints}{output from \code{\link[strucchangeRcpp]{breakpoints}},
+#' containing information about the estimated breakpoints.}
+#' \item{data_pp}{preprocessed data as output by \code{\link{bfastpp}}.}
+#' @example examples/bfastlite.r
 #'
-#' @export bfast0n
+#' @export
 
-bfast0n <- function(data, formula=response ~ trend + harmon, order = 3,
+bfastlite <- function(data, formula=response ~ trend + harmon, order = 3,
                    lag = NULL, slag = NULL, na.action = na.omit,
                    stl = c("none", "trend", "seasonal", "both"),
                    decomp=c("stlplus", "stl"), sbins=1, ...)
 {
   data_pp <- bfastpp(data, order = order,
                    lag = lag, slag = slag, na.action = na.action,
                    stl = stl, decomp=decomp, sbins=sbins)
-  breakpoints <- strucchange::breakpoints(formula=formula, data = data_pp, ...)
-  return(breakpoints)
+  breakpoints <- strucchangeRcpp::breakpoints(formula=formula, data = data_pp, ...)
+  Result <- structure(list(breakpoints, data_pp), class="bfastlite")
```

Comment:
I don't like that Result starts with a capital R. So I checked how it is in all the other "results" and it's a mess. I'll start a new issue about this issue.